### PR TITLE
"open-libra genesis" subcommand

### PR DESCRIPTION
**Pull request by tarcieri**
Adds a `genesis` subcommand which takes a set of `peer_info.toml`-formatted files and generates a random `mint.key`, then computes `genesis.blob`, `consensus_peers.config.toml`, and `network_peers.config.toml`

TODO: separate faucet keygen command

Edit: Removed WIP. We were able to successfully boot a Libra cluster using this tool

### CODE REVIEW

File: src/commands/genesis.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+        let mint_key_file = self.output_dir_or_default().join("mint.key");
+        File::create(&mint_key_file)?.write_all(&bincode::serialize(&keypair).unwrap())?;
+        status_ok!("Generated", "{}", mint_key_file.display());
+
+        Ok(keypair)
+    }
+
+    /// Get the output directory or the default
+    fn output_dir_or_default(&self) -> &Path {
```
Comment (liangping) [inline src/commands/genesis.rs]: also need check whether or not the output_dir exits.

Comment (tarcieri) [reply src/commands/genesis.rs]: The upstream `libra-config` tool does `std::fs::create_dir_all` on the `output_dir`. Perhaps we should do the same.

**Comment by tarcieri** [comment]
Removed WIP.

I still need to add a `seed_peers.config.toml` generator, but otherwise this is ready for review.

**Comment by tarcieri** [comment]
Just added support for generating `seed_peers.config.toml`. Will merge when CI is green.

**Pull request merged**